### ISSUE TITLE
Turns off the light.

### ISSUE THREAD

**Issue by edsallo** [lvanderree/com.synplyworks.thenmore#3]
If the motion constantly and did not stop while the timer is running, the light still turns off. That is, if the sensor did not go out of the so-called "motion alarm turned on", then the light will turn off.

**Comment by lvanderree** [comment]
I am afraid the timer app cannot fix this issue. Your motion detector probably doesn't send new motion events, so the flow does not get triggered again, so the timer does not get reset.

Maybe you can configure your motion sensor to (re)send events during motion, or maybe you can change your flows: turn on the light when motion detected, turn off light after motion sensor sends event that motion has stopped (if it sends out that event)

**Comment by lvanderree** [comment]
I understand what you say/want but unfortunately Homey does not work that way. The app provides an action card that (re)starts an timer to turn on and afterwards turn off a device. 

It  provides timers, and the possibility to read out of timers are running, and the option to cancel timers. 
It cannot listen to events of the the trigger (your motion sensor). That is the way Homey is setup. You should design that yourself in flows.

You can however configure your motion sensor to report about motion that is still going on (at least the Coolcams can do that, I use those myself)

**Comment by edsallo** [comment]
Thank!
What parameters should I change?
On the example of Coolcam.
[http://manuals-backend.z-wave.info/make.php?lang=en&sku=NAS-PD01ZE&cert=ZC10-16105298](url)

**Comment by lvanderree** [comment]
What I did is set a very short time for the blind time param 2 (1s) and a short time for motion detection param 6 (5s).
This isn't perfect, but the easiest to make it work fairly ok. 

I am however thinking about a solution to make it work better, but that will require 2 flows: first to 'prepare' the timer (turning lights on, and remember previous state) on motion detected, the second to start the timer (which will restore the device to previous state when timer ends) on motion ended trigger

**Comment by edsallo** [comment]
![3 flow - homey 2018-12-14 04-17-17](https://user-images.githubusercontent.com/35398195/49977448-7251c800-ff57-11e8-9174-e69a1ab6214d.png)
![3 flow - homey 2018-12-14 04-18-11](https://user-images.githubusercontent.com/35398195/49977465-7f6eb700-ff57-11e8-823b-890ee27e5133.png)

**Comment by lvanderree** [comment]
Exactly,those two triggers. What you can do for now, that will probably result in the same effect is: replace the 'turn light on' with a start timer, configured with a very long timeout. And configure the second flow with the timer to reset the timeout, even when the new timeout is shorter. After that timeout it will restore the lights to the state before the motion detected if I made it correctly

**Comment by edsallo** [comment]
In the first flow, after turning on the light, it is necessary to add switching off the timer, then it works perfectly.
![screenshot_14](https://user-images.githubusercontent.com/35398195/50000681-57a93e80-ffab-11e8-942c-56103086ad01.jpg)

**Comment by lvanderree** [comment]
That is another way of doing it yes. it looks like a pretty stable solution this way. Only a petty that you still need 2 flows, but I haven't thought about any other solution with flows to fix that. The only thing I can think of is pairing all motion detectors, and devices in a settings page, but I don't want to diverge too much from the Homey way of doing things.

Ps. For Homey 2 I've updated the app to cancel running timers when you manually turn off the device (pressing the light button for instance). With that updated you probably don't need the cancel-timer card anymore, or why do you need the cancel timer cards for?

If you like I can backport these changes for Homey 1.5 as well, but I don't want to release those to the appstore before it is being tested (and since I am running Homey 2, I cannot test it anymore). So if you are comfortable with pulling a release from git and sideload it yourself with athom-cli I can make a new  1.5 version if you like.

**Comment by edsallo** [comment]
If there was a report "Motion Off", then there was a movement and did not stop, the timer will turn off the light.
For this in the first flow - "cancel timer"
Yes, I can install through the CLI.

2.0 was too buggy and I returned to 1.5.
